work_queue/src/work_queue.c

```diff
@@ -955,13 +996,18 @@ static void add_worker(struct work_queue *q)
 	w->current_tasks_boxes = itable_create(0);
 	w->finished_tasks = 0;
 	w->start_time = timestamp_get();
-
+	
+	w->worker_link_port = -1;
+	
 	w->last_update_msg_time = w->start_time;
 
 	w->resources = work_queue_resources_create();
 
 	w->workerid = NULL;
 
+	w->is_fetching = 0;
```

Comment:
this member variable is not necessary, as when a worker is not involved in a fetch then w->ongoing_fetch == NULL. Will remove in the near future